framework/src/org/checkerframework/common/value/ValueAnnotatedTypeFactory.java

```diff
@@ -1527,12 +1697,38 @@ public AnnotationMirror convertArrayLenToArrayLenRange(AnnotationMirror arrayLen
         return createArrayLenRangeAnnotation(Collections.min(values), Collections.max(values));
     }
 
+    /** Converts an {@code @StringVal} annotation to an {@code @ArrayLenRange} annotation. */
+    public AnnotationMirror convertStringValToArrayLenRange(AnnotationMirror stringValAnno) {
+        List<String> values = getStringValues(stringValAnno);
+        List<Integer> lengths = ValueCheckerUtils.getLengthsForStringValues(values);
+        return createArrayLenRangeAnnotation(Collections.min(lengths), Collections.max(lengths));
+    }
+
+    /** Converts an {@code @StringVal} annotation to an {@code @ArrayLen} annotation. */
+    public AnnotationMirror convertStringValToArrayLen(AnnotationMirror stringValAnno) {
+        List<String> values = getStringValues(stringValAnno);
+        return createArrayLenAnnotation(ValueCheckerUtils.getLengthsForStringValues(values));
+    }
+
```

Comment:
Why the distinction between `convertStringValToArrayLen` and `convertStringValToArrayLenRange`? A better design might be to have one method that accepts a `StringVal` and returns either an `ArrayLen` or an `ArrayLenRange` as appropriate. Then you don't have to do a check everytime these are called (and future programmers won't either).